tests/test_tree.rs

```diff
@@ -162,4 +159,128 @@ fn recovery() {
+        if g.gen_weighted_bool(50) {
+            return Op::Restart;
+        }
+
+        let choice = g.gen_range(0, 5);
+
+        match choice {
+            0 => Op::Set(g.gen::<u8>(), g.gen::<u8>()),
+            1 => Op::Get(g.gen::<u8>()),
+            2 => Op::Del(g.gen::<u8>()),
+            3 => Op::Cas(g.gen::<u8>(), g.gen::<u8>(), g.gen::<u8>()),
+            4 => Op::Scan(g.gen::<u8>(), g.gen_range::<usize>(0, 40)),
+            _ => panic!("impossible choice"),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct OpVec {
+    ops: Vec<Op>,
+}
+
+impl Arbitrary for OpVec {
+    fn arbitrary<G: Gen>(g: &mut G) -> OpVec {
+        let mut ops = vec![];
+        for _ in 0..g.gen_range(1, 100) {
+            let op = Op::arbitrary(g);
+            ops.push(op);
+
+        }
+        OpVec {
+            ops: ops,
+        }
+    }
+
+    fn shrink(&self) -> Box<Iterator<Item = OpVec>> {
+        let mut smaller = vec![];
+        for i in 0..self.ops.len() {
+            let mut clone = self.clone();
+            clone.ops.remove(i);
+            smaller.push(clone);
+        }
+
+        Box::new(smaller.into_iter())
+    }
+}
+
+fn prop_tree_matches_btreemap(ops: OpVec) -> bool {
+    use self::Op::*;
+    let path = Some("qc_tree_matches_btreemap".to_owned());
+    let config = Config::default().blink_fanout(2).path(path);
+    let mut tree = config.tree();
+    let mut reference = BTreeMap::new();
+
+    for op in ops.ops.into_iter() {
+        match op {
+            Set(k, v) => {
+                tree.set(vec![k], vec![v]);
+                reference.insert(k, v);
+            }
+            Get(k) => {
+                let res1 = tree.get(&*vec![k]).map(|v_vec| v_vec[0]);
+                let res2 = reference.get(&k).cloned();
+                assert_eq!(res1, res2);
+            }
+            Del(k) => {
+                tree.del(&*vec![k]);
+                reference.remove(&k);
+            }
+            Cas(k, old, new) => {
+                let tree_old = tree.get(&*vec![k]);
+                if tree_old == Some(vec![old]) {
+                    tree.set(vec![k], vec![new]);
+                }
+
+                let ref_old = reference.get(&k).cloned();
+                if ref_old == Some(old) {
+                    reference.insert(k, new);
+                }
+            }
+            Scan(k, len) => {
+                let tree_iter = tree.scan(&*vec![k]).take(len).map(|(ref tk, ref tv)| {
+                    (tk[0], tv[0])
+                });
+                let ref_iter = reference
+                    .iter()
+                    .filter(|&(ref rk, _rv)| **rk >= k)
+                    .take(len)
+                    .map(|(ref rk, ref rv)| (**rk, **rv));
+                for (t, r) in tree_iter.zip(ref_iter) {
+                    assert_eq!(t, r);
+                }
+            }
+            Restart => {
+                drop(tree);
+                tree = config.tree();
+            }
```

Comment:
for each operation, we apply the same thing to the `BTreeMap` (except restart, which exercises the recovery logic). For things that return results, we ensure that both things return the same thing. If our tree returns something else, there's a bug somewhere.